core/api/serializers/business_plan.py

```diff
@@ -313,29 +368,19 @@ class Meta:
             "comment_types",
             "values",
         ]
+        list_serializer_class = BulkCreateListSerializer  # save objs in bulk
 
-    def _create_bp_activity_values(self, bp_activity, activity_values):
-        bp_activity.values.all().delete()
-
-        for activity_value in activity_values:
-            activity_value["bp_activity_id"] = bp_activity.id
-        activity_value_serializer = BPActivityValueSerializer(
-            data=activity_values, many=True
-        )
-        activity_value_serializer.is_valid(raise_exception=True)
-        activity_value_serializer.save()
-
-    @transaction.atomic
+    # don't call `save()` here
     def create(self, validated_data):
-        if self.context.get("ignore_comment", False):
-            validated_data.pop("comment_secretariat", "")
-            validated_data.pop("comment_types", [])
+        data = validated_data.copy()
+        # remove m2m fields
+        data.pop("values", [])
```

Comment:
I don't think it's ideal to drop the attributes here and create the objects in the parent serializer. 
While this approach might be faster, the logic is more complex and could be harder to maintain in the future. 
A better approach would be to drop the list attributes before creating the activity, and then handle them together in the same serializer after activity creation.